### PR TITLE
Features/assess page data impact db side

### PR DESCRIPTION
**Pull request by SamirHassoun**
1) Added DataImpact and HpdUsage tables
2) Removed Folder option from factory
3) Added Data Impact partial to Assess page
4) Added Wait modal and error popup when Assess Done is clicked

### CODE REVIEW

File: src/Portal/Portal/Pages/DbAssessment/Assess.cshtml [hunk 1]
```diff
@@ -96,4 +101,55 @@
...
+            </div>
+        </div>
+    </div>
+
+</div>
+
+<div class="modal fade" id="modalWaitAssessDoneErrors" tabindex="-1" role="dialog" aria-labelledby="OnHoldError" aria-hidden="true">
+    <div class="modal-dialog error ">
+        <div class="modal-content">
+            <form id="frmAssessDoneError">
```
Comment (UKHOGreg) [inline src/Portal/Portal/Pages/DbAssessment/Assess.cshtml]: Do we need a form here?

File: src/Portal/Portal/Pages/DbAssessment/Assess.cshtml [hunk 2]
```diff
@@ -96,4 +101,55 @@
...
+                                <ul>
+                                    @foreach (var message in Model.ValidationErrorMessages)
+                                    {
+                                        <li>@message</li>
+                                    }
+                                </ul>
+                            }
+                        </div>
+                    </div>
+                    <input type="hidden" asp-for="ProcessId" />
```
Comment (UKHOGreg) [inline src/Portal/Portal/Pages/DbAssessment/Assess.cshtml]: Do we need this hidden field for ProcessId?

File: src/Portal/Portal/Pages/DbAssessment/Assess.cshtml [hunk 3]
```diff
@@ -96,4 +101,55 @@
...
+                    <li>Verifying Data...</li>
+                    <li>Saving Data...</li>
+                </ul>
+            </div>
+        </div>
+    </div>
+
+</div>
+
+<div class="modal fade" id="modalWaitAssessDoneErrors" tabindex="-1" role="dialog" aria-labelledby="OnHoldError" aria-hidden="true">
```
Comment (UKHOGreg) [inline src/Portal/Portal/Pages/DbAssessment/Assess.cshtml]: > aria-labelledby="OnHoldError"
Let's rename that 🙂

File: src/Portal/Portal/wwwroot/js/Assess.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+﻿$(document).ready(function() {
+    setAssessDoneHandler();
+
+    if ($("#assessDoneErrorMessage").html().trim().length > 0) {
+        $("#modalWaitAssessDoneErrors").modal("show");
+    }
+});
+
+function setAssessDoneHandler() {
+    $("#btnDone").prop("disabled", false);
```
Comment (benbhall) [inline src/Portal/Portal/wwwroot/js/Assess.js]: Need to start looking at caching some of our DOM searches to watch JS DOM traversals don't start to slow down our pages too much.

i..e. var assessDone = $(#assesDone)

Leave it today. Perhaps we can go through all the JS at a later date.

**Comment by Rossall82** [comment]
> Think we are almost there.
> 
> We also need to update the DatabaseDesign.md.

And TM Replacement HLD!

File: src/Portal/Portal/Pages/DbAssessment/Assess.cshtml (hunk 1 above)
Comment (SamirHassoun) [reply src/Portal/Portal/Pages/DbAssessment/Assess.cshtml]: didn't we talk about that yesterday?

File: src/Portal/Portal/Pages/DbAssessment/Assess.cshtml (hunk 2 above)
Comment (SamirHassoun) [reply src/Portal/Portal/Pages/DbAssessment/Assess.cshtml]: didn't we talk about that yesterday?

File: src/Common.Helpers/DatabasesHelpers.cs [hunk 5]
```diff
@@ -42,6 +42,8 @@ public static void ClearWorkflowDbTables(WorkflowDbContext workflowDbContext)
             workflowDbContext.Database.ExecuteSqlCommand("delete from [OnHold]");
             workflowDbContext.Database.ExecuteSqlCommand("delete from [TaskNote]");
             workflowDbContext.Database.ExecuteSqlCommand("delete from [DatabaseDocumentStatus]");
+            workflowDbContext.Database.ExecuteSqlCommand("delete from [HpdUsage]");
+            workflowDbContext.Database.ExecuteSqlCommand("delete from [DataImpact]");
```
[the remark this answers is not in the document]
Comment (SamirHassoun) [reply src/Common.Helpers/DatabasesHelpers.cs]: fixed, thanks

File: src/Portal/Portal/Pages/DbAssessment/Assess.cshtml (hunk 3 above)
Comment (SamirHassoun) [reply src/Portal/Portal/Pages/DbAssessment/Assess.cshtml]: Fixed, thanks

File: src/Portal/Portal/Pages/DbAssessment/Assess.cshtml (hunk 2 above)
Comment (SamirHassoun) [reply src/Portal/Portal/Pages/DbAssessment/Assess.cshtml]: Fixed, thanks

Comment (benbhall) [reply src/Portal/Portal/Pages/DbAssessment/Assess.cshtml]: Perhaps not as we have: 

`<input type="hidden" id="hdnProcessId" value="@Model.ProcessId" />` being used by the JS calls.

File: src/Portal/Portal/Pages/DbAssessment/Assess.cshtml (hunk 1 above)
Comment (SamirHassoun) [reply src/Portal/Portal/Pages/DbAssessment/Assess.cshtml]: Fixed, thanks

File: src/Portal/Portal/Pages/DbAssessment/Assess.cshtml (hunk 2 above)
Comment (UKHOGreg) [reply src/Portal/Portal/Pages/DbAssessment/Assess.cshtml]: Think we have always been relying on a hdnProcessId on the assess page itself.
This one is being made inside an error modal, so not sure it should be there.

File: src/Portal/Portal/Pages/DbAssessment/Assess.cshtml (hunk 1 above)
Comment (UKHOGreg) [reply src/Portal/Portal/Pages/DbAssessment/Assess.cshtml]: Forgive me, but I don't see this change?

Comment (UKHOGreg) [reply src/Portal/Portal/Pages/DbAssessment/Assess.cshtml]: Done

**Pull request merged**